### PR TITLE
[Model Runner V2] Support num NaNs in logits

### PR DESCRIPTION
**Pull request by WoosukKwon**

### CODE REVIEW

File: vllm/v1/worker/gpu/async_utils.py [hunk 1]
```diff
@@ -54,6 +54,10 @@ def __init__(
                 )
             else:
                 self.logprobs_tensors = None
+            if sampler_output.num_nans is not None:
+                self.num_nans = sampler_output.num_nans.to("cpu", non_blocking=True)
```
Comment (njhill) [inline vllm/v1/worker/gpu/async_utils.py]: As discussed, could do the `numpy()` here, same for `num_sampled_tokens_np`

**Review by njhill — APPROVED** [review]

File: vllm/v1/worker/gpu/metrics/logits.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    BLOCK_SIZE = 8192
+    num_nans = torch.empty(num_reqs, dtype=torch.int32, device=logits.device)
+    _num_nans_kernel[(num_reqs,)](
+        logits,
+        logits.stride(0),
+        num_nans,
+        vocab_size,
+        BLOCK_SIZE=BLOCK_SIZE,
+    )
+    return num_nans
```
[the remark this answers is not in the document]
Comment (WoosukKwon) [reply vllm/v1/worker/gpu/metrics/logits.py:42]: Good point. I considered it, but wanted to use this Triton version because the performance improvement is substantial. Also, I think this metric could be turned on by default in the future; it's a reasonable metric to track.

File: vllm/v1/worker/gpu/sample/sampler.py [hunk 3]
```diff
@@ -42,13 +45,15 @@ def __call__(
         else:
             logprobs_tensors = None
 
+        num_nans = get_num_nans(processed_logits) if self.compute_nans else None
```
[the remark this answers is not in the document]
Comment (WoosukKwon) [reply vllm/v1/worker/gpu/sample/sampler.py]: Good point. Updated the code to follow the original definition.

File: vllm/v1/worker/gpu/async_utils.py (hunk 1 above)
Comment (WoosukKwon) [reply vllm/v1/worker/gpu/async_utils.py]: Good catch. Fixed!
